src/daemon/rpmostreed-daemon.c

```diff
@@ -332,12 +339,39 @@ on_name_owner_changed (GDBusConnection  *connection,
     rpmostreed_daemon_remove_client (self, name);
 }
 
+static gboolean
+idle_update_status (void *data)
+{
+  RpmostreedDaemon *self = data;
+
+  update_status (self);
+
+  return TRUE;
+}
+
+
+static gboolean
+on_idle_exit (void *data)
+{
+  RpmostreedDaemon *self = data;
+
+  g_clear_pointer (&self->idle_exit_source, (GDestroyNotify)g_source_unref);
+
+  sd_notifyf (0, "STOPPING=1\nSTATUS=Exiting due to idle");
+  self->running = FALSE;
+  g_main_context_wakeup (NULL);
```

Comment:
Yep, we don't need it here.  However, if one is doing multithreading, the main loop does need to be woken up.  So I always add this as part of the "manage mainloop termination condition" pattern.  It's just a `write()` to an `eventfd()` underneath so it's cheap.